clients/v1_repl/v1_repl.go

```diff
@@ -0,0 +1,759 @@
+	}
+	return nil
+}
+
+func (c *Client) gopher() {
+	color.Cyan(Gopher)
+}
+
+// The logic for the main prompt that is run in the REPL loop
+func (c *Client) executor(cmd string) {
+	if cmd == "" {
+		return
+	}
+	cmdArgs := strings.Split(cmd, " ")
+	switch strings.ToLower(cmdArgs[0]) {
+	case "quit", "exit":
+		color.HiBlack("Goodbye!")
+		os.Exit(0)
+	case "gopher":
+		c.gopher()
+	case "node":
+		color.Yellow("The 'node' command is enterprise only, not available in the influx 2.x CLI - were you looking for the 1.x InfluxDB CLI?")
+	case "consistency":
+		color.Yellow("The 'consistency' command is not available in the influx 2.x CLI - were you looking for the 1.x InfluxDB CLI?")
+	case "help":
+		c.help()
+	case "history":
+		color.Yellow("The 'history' command is not yet implemented in 2.x")
+	case "format":
+		c.setFormat(cmdArgs)
+	case "precision":
+		c.setPrecision(cmdArgs)
+	case "settings":
+		c.settings()
+	case "pretty":
+		c.togglePretty()
+	case "use":
+		c.use(cmdArgs)
+	case "insert":
+		c.insert(cmd)
+	case "clear":
+		c.clear(cmd)
+	default:
+		c.runAndShowQuery(cmd)
+	}
+}
+
+// Create a regex string for a named InfluxQL identifier, quoted or unquoted
+func identRegex(name string) string {
+	return `((?P<` + name + `>\w+)|(\"(?P<` + name + `_quote>.+)\"))`
+}
+
+// Get the value of a named InfluxQL identifier from a regex match map.
+// Returns empty string if no match
+func getIdentFromMatches(matches *map[string]string, name string) string {
+	if val, ok := (*matches)[name]; ok && val != "" {
+		return val
+	} else if val, ok := (*matches)[name+"_quote"]; ok && val != "" {
+		return val
+	}
+	return ""
+}
+
+// Create a regex match map from a regexp with named groups.
+// Returns nil if no match.
+func reSubMatchMap(r *regexp.Regexp, str string) *map[string]string {
+	match := r.FindStringSubmatch(str)
+	if match == nil {
+		return nil
+	}
+	subMatchMap := make(map[string]string)
+	for i, name := range r.SubexpNames() {
+		if i != 0 {
+			subMatchMap[name] = match[i]
+		}
+	}
+	return &subMatchMap
+}
+
+// the (?i) clause makes the regex match case-insensitive
+var insertIntoRegex string = `^(?i)INSERT(\s+)INTO(\s+)` + identRegex("db") + `(\.` + identRegex("rp") + `)?(\s+)(?P<point>.+)$`
+var insertRegex string = `^(?i)INSERT(\s+)(?P<point>.+)$`
+
+func (c Client) insert(cmd string) {
+	var db string
+	var rp string
+	var point string
+	insertRgx := regexp.MustCompile(insertRegex)
+	insertIntoRgx := regexp.MustCompile(insertIntoRegex)
+	insertMatches := reSubMatchMap(insertRgx, cmd)
+	insertIntoMatches := reSubMatchMap(insertIntoRgx, cmd)
+	if insertIntoMatches != nil {
+		db = getIdentFromMatches(insertIntoMatches, "db")
+		rp = getIdentFromMatches(insertIntoMatches, "rp")
+		point = getIdentFromMatches(insertIntoMatches, "point")
+	} else if !insertIntoRgx.Match([]byte(cmd)) && insertMatches != nil {
+		db = c.Database
+		rp = c.RetentionPolicy
+		point = getIdentFromMatches(insertMatches, "point")
```

Comment:
I think I like the little parser we had before better, but this works so 👍 .

Reply:
Could you extract this parsing logic (going from `cmd` to `db, rp, point`) into a function, and write a unit test for it? That way we can make sure the various things we discussed continue to work:
* whitespace
* case insensitive

Reply:
Added a unit test file.